src/AppInstallerRepositoryCore/ARPCorrelation.cpp

```diff
@@ -0,0 +1,442 @@
+    {
+        InstanceInternal(algorithmOverride);
+    }
+
+    void ARPCorrelationAlgorithm::ResetInstance()
+    {
+        InstanceInternal(nullptr);
+    }
+#endif
+
+    double NormalizedNameAndPublisherCorrelationMeasure::GetMatchingScore(std::string_view packageName, std::string_view packagePublisher, std::string_view arpName, std::string_view arpPublisher) const
+    {
+        NameNormalizer normer(NormalizationVersion::Initial);
+
+        auto packageNormalizedName = normer.Normalize(packageName, packagePublisher);
+        auto arpNormalizedName = normer.Normalize(arpName, arpPublisher);
+
+        if (Utility::CaseInsensitiveEquals(arpNormalizedName.Name(), packageNormalizedName.Name()) &&
+            Utility::CaseInsensitiveEquals(arpNormalizedName.Publisher(), packageNormalizedName.Publisher()))
+        {
+            return 1;
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    double EditDistanceNormalizedNameAndPublisherCorrelationMeasure::GetMatchingScore(std::string_view packageName, std::string_view packagePublisher, std::string_view arpName, std::string_view arpPublisher) const
+    {
+        NameNormalizer normer(NormalizationVersion::Initial);
+
+        auto packageNormalizedName = normer.Normalize(packageName, packagePublisher);
+        auto arpNormalizedName = normer.Normalize(arpName, arpPublisher);
+
+        auto nameDistance = EditDistanceScore(arpNormalizedName.Name(), packageNormalizedName.Name());
+        auto publisherDistance = EditDistanceScore(arpNormalizedName.Publisher(), packageNormalizedName.Publisher());
+
+        // TODO: Consider other ways of merging the two values
+        return (2 * nameDistance + publisherDistance) / 3;
+    }
+
+    std::shared_ptr<AppInstaller::Repository::IPackageVersion> FindARPEntryForNewlyInstalledPackage(
+        const Manifest::Manifest& manifest,
+        const std::vector<ARPEntrySnapshot>& arpSnapshot,
+        Source& arpSource,
+        std::string_view sourceIdentifier)
+    {
+        AICLI_LOG(Repo, Verbose, << "Finding ARP entry matching newly installed package");
+        std::vector<ResultMatch> changes;
+
+        for (auto& entry : arpSource.Search({}).Matches)
+        {
+            auto installed = entry.Package->GetInstalledVersion();
+
+            if (installed)
+            {
+                auto entryKey = std::make_tuple(
+                    entry.Package->GetProperty(PackageProperty::Id),
+                    installed->GetProperty(PackageVersionProperty::Version),
+                    installed->GetProperty(PackageVersionProperty::Channel));
+
+                auto itr = std::lower_bound(arpSnapshot.begin(), arpSnapshot.end(), entryKey);
+                if (itr == arpSnapshot.end() || *itr != entryKey)
+                {
+                    changes.emplace_back(std::move(entry));
+                }
+            }
+        }
+
+        // Also attempt to find the entry based on the manifest data

```

Comment:
I would like this older matching mechanism to fit into the same infrastructure so that the overall could be something like:
```
foreach (algorithm : { ... })
{
   algorithm.Init(manifest)
   if (Correlate(algorithm, ARP) > threshold)
       return that;
}
```
Thoughts for the future, doesn't need to change now.